engine/common/rpc/errors.go

```diff
@@ -20,16 +20,16 @@ func ConvertError(err error, msg string, defaultCode codes.Code) error {
 		return nil
 	}
 
-	// Already converted
-	if status.Code(err) != codes.Unknown {
-		return err
-	}
-
 	// Handle multierrors separately
 	if multiErr, ok := err.(*multierror.Error); ok {
 		return ConvertMultiError(multiErr, msg, defaultCode)
 	}
 
+	// Already converted
+	if status.Code(err) != codes.Unknown {
+		return err
+	}
```

Comment:
Thanks. This is the correct fix. There was a [change](https://github.com/grpc/grpc-go/compare/v1.53.x...v1.55.x#diff-2b43f8f754e7e61ebc349444f31f588e20353ee7e69b174c7cfbaad101a6d0ebR93-R101) to the grpc library, which results in `status.Code(err)` returning the status of the first error in a multierror. Moving this check to after handling the multierror allows us to use our own logic to handle them.